examples/ssh/opkssh/cli.go

```diff
@@ -0,0 +1,308 @@
+		return nil, nil, err
+	}
+
+	sshcert, err := cert.SignCert(mas)
+
+	if err != nil {
+		return nil, nil, err
+	}
+	certBytes := ssh.MarshalAuthorizedKey(sshcert)
+
+	seckeySsh, err := ssh.MarshalPrivateKey(signer, "openpubkey cert")
+	if err != nil {
+		return nil, nil, err
+	}
+	seckeySshBytes := pem.EncodeToMemory(seckeySsh)
+
+	return certBytes, seckeySshBytes, nil
+}
+
+func WriteKeys(seckeyPath string, pubkeyPath string, seckeySshPem []byte, certBytes []byte) error {
+	// Write ssh secret key to filesystem
+	err := os.WriteFile(seckeyPath, seckeySshPem, 0600)
+	if err != nil {
+		return err
+	}
+	certBytes = append(certBytes, []byte(" "+"openpubkey")...)
+
+	// Write ssh public key (certificate) to filesystem
+	err = os.WriteFile(pubkeyPath, certBytes, 0777)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func FileExists(fPath string) bool {
+	_, error := os.Open(fPath)
+	return !errors.Is(error, os.ErrNotExist)
+}
+
+func WriteKeysToSSHDir(seckeySshPem []byte, certBytes []byte) error {
+	homePath, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+	sshPath := filepath.Join(homePath, ".ssh")
+
+	//  To enable ssh to automatically find the key created by openpubkey we
+	// need to use a default ssh key path. However this means that this
+	// filename might already be in use by the user. To ensure we don't
+	// overwrite a ssh key not created by openpubkey we check the comment in the
+	// key to see if it was created by openpubkey
+	defaultKeyNames := []string{"id_ecdsa_sk", "id_ecdsa", "id_dsa"}
+	for i := range []string{"id_ecdsa_sk", "id_ecdsa", "id_dsa"} {
+		seckeyPath := filepath.Join(sshPath, defaultKeyNames[i])
+		pubkeyPath := seckeyPath + ".pub"
+
+		if !FileExists(seckeyPath) {
+			// If ssh key file does not currently exist, we don't have to worry about overwriting it
+			return WriteKeys(seckeyPath, pubkeyPath, seckeySshPem, certBytes)
+		} else if !FileExists(pubkeyPath) {
+			continue
+		} else {
+			// If ssh key does file does exist, check if it is an openpubkey file, if it is then it is safe to overwrite
+			sshPubkey, err := os.ReadFile(pubkeyPath)
+			if err != nil {
+				fmt.Println("Failed to read:", pubkeyPath)
+				continue
+			}
+			sshPubkeyTok := strings.Split(string(sshPubkey), " ")
+			if len(sshPubkeyTok) != 3 {
+				fmt.Println("Failed to parse:", pubkeyPath)
+				continue
+			}
+			// check if pubkey comment to see if it an openpubkey ssh key
+			if strings.Contains(sshPubkeyTok[2], ("openpubkey")) {
+				// safe to overwrite
+				return WriteKeys(seckeyPath, pubkeyPath, seckeySshPem, certBytes)
+			}
+		}
+	}
+	return fmt.Errorf("no default ssh key file free for openpubkey")
+}
+
+func log(line string) {
+	f, err := os.OpenFile("/var/log/openpubkey.log", os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0700)
+	if err != nil {
+		fmt.Println("Couldn't write to file")
+	} else {
+		defer f.Close()
+		if _, err = f.WriteString(line + "\n"); err != nil {
+			fmt.Println("Couldn't write to file")
+		}
+	}
```

Comment:
I don't think your change here works.
if err!=nil and then you still try to write to the file. The else prevents you from writing to the file if the file can't be opened.

Reply:
oh, you're right, I just assumed this function was returning errors. Since you're intent on this pattern you can resolve, but I don't think it's a good one